### ISSUE TITLE
New version of raster

### ISSUE THREAD

**Issue by AugustT** [zoonproject/zoon#423]
The new version of Raster causes an error in testing. 

**TL;DR - I need to tweak one of the tests**

Conversation with raster maintainer:

****

Dear Tom,
 
The recent update of the R package "raster" causes a test error (due to an error not happening!) in your package "zoon". This happens in the "workflow" function, which is a bit trickier than other functions to understand and debug. See:
 
https://cran.r-project.org/web/checks/check_results_zoon.html
 
Could you have a look at let me know if this is due an error in "raster"?
 
Thanks, Robert

****

Hi Robert,
 
I *think* I have traced the change in behaviour of raster that is causing this.
 
In this scenario I have a raster I want to extract data from using points. I have subset my points in a way that returns a dataframe with 0 rows (assume I did something stupid). I then run raster::extract. Repex attached. I believe that this used to return NULL but now returns ‘numeric(0)’. This is causing zoon problems because it uses ‘is.null’ to catch this case and report an error and prevent the modelling progressing further.
 
Can you please confirm this change in behaviour of raster? Do you intend to keep it the way it is now? If so I will make changes to zoon’s catch to accommodate this change.
 
Tom

****

Hi Tom, 
Thanks for your quick reply. I found what causes this. In version 2.8-1 I have made a change so that it returns NULL again. However, may I suggest that you change your test to 
if (length(x) == 0) 
As that would work for both cases (NULL and and an empty vector)?
Thanks,
Robert

**Comment by timcdlucas** [comment]
Great work dealing with this so quickly.

Tim

**Comment by AugustT** [comment]
I havent actually implemented the fix yet!
________________________________

**Comment by timcdlucas** [comment]
Oh ok! Well great work... handling it thus far?

**Comment by timcdlucas** [comment]
Seems like tests are passing so I assume this got fixed.
